src/coreclr/tools/dotnet-pgo/Program.cs

```diff
@@ -1382,8 +1391,67 @@ private int InnerProcessTraceFileMain()
 
-                        if ((e.TimeStampRelativeMSec >= excludeEventsBefore) && (e.TimeStampRelativeMSec <= excludeEventsAfter))
+                        if (e.TimeStampRelativeMSec < excludeEventsBefore)
+                        {
+                            continue;
+                        }
+
+                        if (e.TimeStampRelativeMSec > excludeEventsAfter)
+                        {
+                            break;
+                        }
+
+                        string perfviewMethodName = e.MethodNamespace + "." + e.MethodName + paramsArgs;
+                        if (PassesMethodFilter(includeMethods, excludeMethods, perfviewMethodName))
+                        {
                             methodsToAttemptToPrepare.Add((int)e.EventIndex, new ProcessedMethodData(e.TimeStampRelativeMSec, method, "R2RLoad"));
+                        }
+                    }
+                }
+
+                // In case requesting events before/after jitting a method, discover the
+                // corresponding excludeEventsBefore/excludeEventsAfter in event stream based
+                // on filter criterias.
+                if (_command.ProcessJitEvents && (excludeEventsBeforeJittingMethod != null || excludeEventsAfterJittingMethod != null))
+                {
+                    foreach (var e in p.EventsInProcess.ByEventType<MethodJittingStartedTraceData>())
+                    {
+                        if (e.ClrInstanceID != clrInstanceId)
+                        {
+                            continue;
+                        }
+
+                        MethodDesc method = null;
+                        bool failedDueToNonloadableModule = false;
+                        try
+                        {
+                            method = idParser.ResolveMethodID(e.MethodID, out failedDueToNonloadableModule, false);
+                        }
+                        catch { }
+
+                        if (method == null)
+                        {
+                            continue;
+                        }
+
+                        int parenIndex = e.MethodSignature.IndexOf('(');
+                        string paramsArgs = e.MethodSignature.Substring(parenIndex);
+                        string perfviewMethodName = e.MethodNamespace + "." + e.MethodName + paramsArgs;
+                        if (e.TimeStampRelativeMSec > excludeEventsBefore && excludeEventsBeforeJittingMethod != null && excludeEventsBeforeJittingMethod.IsMatch(perfviewMethodName))
+                        {
+                            excludeEventsBefore = e.TimeStampRelativeMSec;
+                        }
+
+                        if (e.TimeStampRelativeMSec < excludeEventsAfter && excludeEventsAfterJittingMethod != null && excludeEventsAfterJittingMethod.IsMatch(perfviewMethodName))
+                        {
+                            excludeEventsAfter = e.TimeStampRelativeMSec;
+                        }
```

Comment:
Is the behavior here on multiple matches the right one? In particular excluding events before the last match instead of before the first match seems odd to me.

Reply:
RIght, it would be problematic especially if the method matching had a timestamp order of `beforeMethodMatch1` < `afterMethodMatch` < `beforeMethodMatch2`. And worst case, if the two matches wer the same, it would be impossible for the user to have a specific enough regex to only match the first (temporally) method.

Are methods in the enumerator of `p.EventsInProcess.ByEventType<MethodJittingStartedTraceData>()` guaranteed to be ordered temporally?

Maybe it would be better to keep a list of the matches and choose the first temporal match for `excludeEventsBeforeJittingMethod` and the last temporal match for `excludeEventsAfterJittingMethod`

Reply:
> Are methods in the enumerator of p.EventsInProcess.ByEventType<MethodJittingStartedTraceData>() guaranteed to be ordered temporally?

Yes, so we can just have a bool or something to keep track of whether we have already assigned `excludeEventsBefore`.

Reply:
And for `excludeEventsAfterJittingMethod` we would need to still keep a list of the matches for a scenario like `afterMethodMatch1` < `beforeMethodMatch` < `afterMethodMatch2` in case the user wanted to use the bounds `[beforeMethodMatch, afterMethodMatch2]`.

Just realizing, if these events are for jitting methods, is it the case that we would only JIT a particular method once? Meaning no two events will have the same name? Not sure exactly how these events are collected. If thats the case, then maybe we can have it such that the user needs to be more specific with their Regex matching if there happens to be multiple matches?

Reply:
I was able to trace an app and sent over the .nettrace to my windows machine. It looks like caching does happen and a method will only be found once in `MethodJittingStartedTraceData`? So it seems like users can be more specific with their regular expression matching, but will opt to take the first "before" match and the last "after" match to avoid invalid bounds.

Reply:
Added two doubles to track the earliest excludeEventsBeforeJittingMethod match and latest excludeEventsAfterJittingMethod match